lib/puppet/provider/package/gem.rb

```diff
@@ -148,12 +170,22 @@ def install(useversion = true)
     command = resource_or_provider_command
     command_options = ["install"]
     command_options += install_options if resource[:install_options]
+    should = resource[:ensure]
+
+    unless should =~ /,/ || should =~ Gem::Requirement::PATTERN
+      begin
+        should_range = Puppet::Util::Package::Version::Range.parse(should, Puppet::Util::Package::Version::Gem)
+        should = should_range.to_s.sub(' ', ',')
```

Comment:
I assume here is the transformation from Version::Range to Gem, For example `>1 <2` to `>1,<2`.
If this it is, it should be part of a Version::Range method, `to_gem_spec` or something, as we should not make here assumptions about `Version::Range` internals.